camayoc/tests/qpc/utils.py

```diff
@@ -93,3 +93,33 @@ def gen_valid_source(cleanup, src_type, hosts, create=True,
         cleanup.append(source)
         assert_matches_server(source)
     return source
+
+
+def sort_and_delete(trash):
+    """Sort and delete a list of QPCObject typed items in the correct order."""
+    creds = []
+    sources = []
+    scans = []
+    # first sort into types because we have to delete scans before sources
+    # and sources before scans
+    for obj in trash:
+        if isinstance(obj, Credential):
+            creds.append(obj)
+            continue
+        if isinstance(obj, Source):
+            sources.append(obj)
+            continue
+        if isinstance(obj, Scan):
+            scans.append(obj)
+            continue
+
+    for collection in [scans, sources, creds]:
+        for obj in collection:
+            # It may have been a while since this object was created, so we
+            # will log its client back in to the server and get a fresh token
+            obj.client.login()
```

Comment:
maybe push this up to line 115 so we don't log in so many times?

Reply:
This is a good call, will update

Reply:
I've created a new client which will ensure it is properly logged in. Then for each object, I am overriding the client to use the newly created one.